File: 백준/Silver/28278. 스택 2/스택 2.cc
#include<iostream>
#include<stack>
using namespace std;

int n, cmd, val;
stack<int>s;

int main()
{
	ios::sync_with_stdio(0); cin.tie(0);
	cin >> n;
	while (n--) {
		cin >> cmd;
		if (cmd == 1) {
			cin >> val;
			s.push(val);
		}
		else if (cmd == 2) {
			if (s.empty()) cout << -1 << '\n';
			else {
				cout << s.top() << '\n';
				s.pop();
			}
		}
		else if (cmd == 3) {
			cout << s.size() << '\n';
		}
		else if (cmd == 4) {
			if (s.empty()) cout << 1 << '\n';
			else cout << 0 << '\n';
		}
		else if (cmd == 5) {
			if (s.empty()) cout << -1 << '\n';
			else cout << s.top() << '\n';
		}
	}

	return 0;
}

File: 백준/Silver/28278. 스택 2/스택 2.java
import java.util.*;
import java.io.*;

class Main {
	public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st;
		
		Stack<Integer> s = new Stack<>();
		
		int n = Integer.parseInt(br.readLine());
		
		while(n-- > 0){
			st = new StringTokenizer(br.readLine());
			int opt = Integer.parseInt(st.nextToken());
			
			if(opt == 1) {
				s.push(Integer.parseInt(st.nextToken()));
			}else if(opt == 2){
				if(s.empty()) System.out.println(-1);
				else{
					int top = s.pop();
					System.out.println(top);
				}
			}else if(opt == 3){
				System.out.println(s.size());
			}else if(opt == 4){
				if(s.empty()) System.out.println(1);
				else System.out.println(0);
			}else{
				if(s.empty()) System.out.println(-1);
				else System.out.println(s.peek());
			}
		}
	}
}